projects/probational/cmd/probational.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// Moby is the type of a Moby config file
+// stolen from github.com/moby/tool/cmd/moby, perhaps we should make handling
+// this into a lib?
+type Moby struct {
+	Kernel struct {
+		Image   string
+		Cmdline string
+	}
+	Init []string
+	//Onboot   []MobyImage
+	//Services []MobyImage
+	//Trust    TrustConfig
+	Files []struct {
+		Path      string
+		Directory bool
+		Symlink   string
+		Contents  string
+		Source    string
+	}
+}
+
+const linuxkitPath = "../linuxkit.yml"
+const kernelSeries = "4.11.x"
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Printf("error: %s", err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	var projectsFile string
+
+	flag.StringVar(&projectsFile, "input", "probational/projects.yml", "The input list of projects")
+	flag.Parse()
+
+	content, err := ioutil.ReadFile(projectsFile)
+	if err != nil {
+		return err
+	}
+
+	var projects []string
+	if err := yaml.Unmarshal(content, &projects); err != nil {
+		return err
+	}
+
+	content, err = ioutil.ReadFile(linuxkitPath)
+	if err != nil {
+		return err
+	}
+
+	os.Remove("kernel-config/kernel_config.probational")
+
+	cfg := Moby{}
+	if err := yaml.Unmarshal(content, &cfg); err != nil {
+		return err
+	}
+
+	for _, p := range projects {
+		if err := addProject(&cfg, p); err != nil {
+			return err
+		}
+	}
+
+	// We need users to build the resulting kernel, so use a dummy in the
+	// yaml output.
+	cfg.Kernel.Image = "your-probational-image"
+
+	raw, err := yaml.Marshal(&cfg)
+	if err != nil {
+		return err
+	}
+
+	if err := ioutil.WriteFile("probational/probational.yml", raw, 0644); err != nil {
+		return err
+	}
+
+	fmt.Println(`Probational configuration complete. Please do:
+
+    cd kernel-config && make IMAGE=probational
+
+The resulting kernel can be used in conjunction with the configuration in
+probational/probational.yml to generate a linuxkit image.
+`)
+
+	return nil
+}
+
+func addProject(cfg *Moby, project string) error {
+	fmt.Printf("adding project %s\n", project)
+
+	content, err := ioutil.ReadFile(filepath.Join(project, fmt.Sprintf("%s.yml", project)))
+	if err != nil {
+		return err
+	}
+
+	projectCfg := Moby{}
+	if err := yaml.Unmarshal(content, &projectCfg); err != nil {
+		return err
+	}
+
+	// merge config
```

Comment:
perhaps (in a future PR?) we can split out this config merging and kernel patches logic into separate, exported functions that can be vendored by other projects?  Might be useful for moby tool and other related code